### PR TITLE
fix: error showing in currency input

### PR DESCRIPTION
**Pull request by techbhavin**
## Description

> Currency Input widget always shows an error when the user writes.

Fixes [#14862](https://github.com/appsmithorg/appsmith/issues/14862)

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Import this application [json](https://github.com/appsmithorg/appsmith/files/8999584/Currency.Input.Issue.json.zip).
- Now try to write inside the currency input, showing the error message.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/cypress/integration/Smoke_TestSuite/Application/CurrencyInputIssue_Spec.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          force: true,
+        });
+        cy.wait(2000);
+      } else {
+        homePage.AssertImport();
+      }
+    });
+  });
+
+  it("2. Should check that widget input show any errors", function() {
```
Comment (sbalaji1192) [inline app/client/cypress/integration/Smoke_TestSuite/Application/CurrencyInputIssue_Spec.js:41]: "should check that the widget input is not showing any error"

File: app/client/cypress/integration/Smoke_TestSuite/Application/CurrencyInputIssue_Spec.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  beforeEach(function() {
+    agHelper.RestoreLocalStorageCache();
+  });
+
+  afterEach(function() {
+    agHelper.SaveLocalStorageCache();
+  });
+
+  it("1. Import application json", function() {
+    cy.visit("/applications");
```
Comment (sbalaji1192) [inline app/client/cypress/integration/Smoke_TestSuite/Application/CurrencyInputIssue_Spec.js:25]: Are we using the same application export we got from the User? we don't need to import such a big application.. you can import this application manually, delete all the widgets and only add a currency widget. Then export and use that as the CurrencyInputIssueExport.json.

File: app/client/cypress/integration/Smoke_TestSuite/Application/CurrencyInputIssue_Spec.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        cy.wait(2000);
+      } else {
+        homePage.AssertImport();
+      }
+    });
+  });
+
+  it("2. Should check that widget input show any errors", function() {
+    cy.get(widgetInput).type("123456789");
+    cy.get(".bp3-popover-content").should("not.exist");
```
Comment (sbalaji1192) [inline app/client/cypress/integration/Smoke_TestSuite/Application/CurrencyInputIssue_Spec.js:43]: can you be more specific with the selector?

File: app/client/cypress/integration/Smoke_TestSuite/Application/CurrencyInputIssue_Spec.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      } else {
+        homePage.AssertImport();
+      }
+    });
+  });
+
+  it("2. should check that the widget input is not showing any error", function() {
+    cy.get(widgetInput).type("123456789");
+    cy.focused().then(() => {
+      cy.get(themelocators.popover).should("not.exist");
```
Comment (sbalaji1192) [inline app/client/cypress/integration/Smoke_TestSuite/Application/CurrencyInputIssue_Spec.js:47]: please check once that this case captures the issue by commenting out the changes we made on the derived property.

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/CurrencyInput_spec.js [hunk 5]
```diff
@@ -272,4 +272,9 @@ describe("Currency widget - ", () => {
     // Check if isDirty is reset to false
     cy.get(".t--widget-textwidget").should("contain", "false");
   });
+
+  it("Should check that widget input show any errors", function() {
```
Comment (sbalaji1192) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/CurrencyInput_spec.js:276]: "Should check that widget input is not showing any errors on input"

File: app/client/cypress/integration/Smoke_TestSuite/Application/CurrencyInputIssue_Spec.js (hunk 4 above)
Comment (techbhavin) [reply app/client/cypress/integration/Smoke_TestSuite/Application/CurrencyInputIssue_Spec.js:47]: Yes, it shows the error.

image

**Review by sbalaji1192 — DISMISSED** [review]

**Comment by chandannkumar** [comment]
@techbhavin 
We can still see the text '**Invalid input**' right after start typing input on currency widget although it disappears automatically
https://www.loom.com/share/65f2bfca926446ba98d3dc947a72c26e

**Comment by chandannkumar** [comment]
Tested this PR and working as expected. Moving to done
Raised callout will get resolved on #7810

**Comment by somangshu** [comment]
@techbhavin there are failing tests here, please check if they need some change or lets merge this soon

**Review by sbalaji1192 — APPROVED** [review]

**Pull request merged**